ferminet/hamiltonian.py

```diff
@@ -144,15 +144,17 @@ def grad_phase_closure(x):
   return _lapl_over_f
 
 
-def potential_electron_electron(r_ee: Array) -> jnp.ndarray:
+def potential_electron_electron(pos: Array) -> jnp.ndarray:
   """Returns the electron-electron potential.
 
   Args:
-    r_ee: Shape (neletrons, nelectrons, :). r_ee[i,j,0] gives the distance
-      between electrons i and j. Other elements in the final axes are not
-      required.
+    pos: Shape (neletrons, ndim). Electron positions.
   """
-  return jnp.sum(jnp.triu(1 / r_ee[..., 0], k=1))
+  nelectrons = pos.shape[0]
+  r_ee = jnp.sqrt(jnp.sum((pos[None, ...] - pos[:, None] + 
+      jnp.eye(nelectrons)[..., None])**2, axis=-1))
+  r_ee = r_ee[jnp.triu_indices_from(r_ee, 1)]
+  return (1./r_ee).sum()
```

Comment:
Yes, you are right! That's the aftermath of my NaNs hunt. You cannot use `jnp.triu(r_ee, k=1)`, though. I've fixed it and have tested it.

Reply:
Ah, I (think I) meant `jnp.sum(jnp.triu(1./(r_ee[..., 0] + jnp.eye(...)), k=1)` but doing `jnp.triu_indices_from` works well (and is probably faster). Either way, the key is to avoid a 1/0 entirely -- by adding the identity to the diagonal (which is then dropped by the `jnp.triu`) or, as you did, by doing the triu before the reciprocal. This is much nicer, thanks!